Makefile

```diff
@@ -1101,15 +1142,29 @@ lint-addon-docs: test/addons/.docbuildstamp
 cpplint: lint-cpp
 	@echo "Please use lint-cpp instead of cpplint"
 
-ifneq ("","$(wildcard tools/eslint/)")
+# tools/eslint is not available in the source tarball
+ifeq ("","$(wildcard tools/eslint/)")
+lint:
+	@echo "Linting is not available through the source tarball."
+	@echo "Use the git repo instead:" \
+		"$ git clone https://github.com/nodejs/node.git"
+
+lint-ci: lint
+else
+# Inside the git repository
+# This runs linters on different type of files. It's supposed to
+# be run by the developers.
 lint:
 	@EXIT_STATUS=0 ; \
 	$(MAKE) lint-js || EXIT_STATUS=$$? ; \
 	$(MAKE) lint-cpp || EXIT_STATUS=$$? ; \
 	$(MAKE) lint-md || EXIT_STATUS=$$? ; \
 	$(MAKE) lint-addon-docs || EXIT_STATUS=$$? ; \
 	exit $$EXIT_STATUS
+
 CONFLICT_RE=^>>>>>>> [0-9A-Fa-f]+|^<<<<<<< [A-Za-z]+
+
+# This is run by the node-test-linter CI job
```

Comment:
General observation: the new comments should be properly punctuated (dot at the end.)